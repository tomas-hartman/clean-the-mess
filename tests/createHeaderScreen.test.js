const { createHeaderScreen } = require("../src/popup/popup");
const tabsOverviewMinified = [{
    "url": "https://www.stackoverflow.com",
    "count": 5,
    "ids": [
      87,
      95,
      94,
      98,
      100
    ]
}];

describe('Create header screens for different "types": ', () => {
    it.each([
        ["latest"],
        ["details"]
    ])(
        'should create correct header for %s',
        (input) => {
            
            const output = createHeaderScreen(0, input, tabsOverviewMinified);

            expect(output).toMatchSnapshot();
        }
    );
});

it('should create correct header for latest without id and tabsOverview', () => {
    const output = createHeaderScreen(null, "latest");

    expect(output).toMatchSnapshot();
});