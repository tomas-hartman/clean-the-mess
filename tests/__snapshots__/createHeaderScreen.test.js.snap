// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create header screens for different "types":  should create correct header for details 1`] = `
Array [
  <DocumentFragment>
    
            
    <div
      id="header"
    >
      
                
      <div
        class="back go-back"
        title="Back"
      />
      
                
      <div
        class="header-title"
      >
        www.stackoverflow.com
      </div>
      
                
      <div
        class="close-all"
        data-index-number="0"
        title="Close all listed tabs"
      />
      
            
    </div>
    
        
  </DocumentFragment>,
  <DocumentFragment>
    <div
      class="separator separator-bottom"
    />
  </DocumentFragment>,
]
`;

exports[`Create header screens for different "types":  should create correct header for latest 1`] = `
Array [
  <DocumentFragment>
    
            
    <div
      id="header"
    >
      
                
      <div
        class="back go-back"
        title="Back"
      />
      
                
      <div
        class="header-title"
      >
        10 longest unused tabs
      </div>
      
                
            
    </div>
    
        
  </DocumentFragment>,
  <DocumentFragment>
    <div
      class="separator separator-bottom"
    />
  </DocumentFragment>,
]
`;

exports[`should create correct header for latest without id and tabsOverview 1`] = `
Array [
  <DocumentFragment>
    
            
    <div
      id="header"
    >
      
                
      <div
        class="back go-back"
        title="Back"
      />
      
                
      <div
        class="header-title"
      >
        10 longest unused tabs
      </div>
      
                
            
    </div>
    
        
  </DocumentFragment>,
  <DocumentFragment>
    <div
      class="separator separator-bottom"
    />
  </DocumentFragment>,
]
`;
